source/ncengine/asset/manager/TextureAssetManager.cpp

```diff
@@ -37,7 +39,41 @@ bool TextureAssetManager::Load(const std::string& path, bool isExternal, asset_f
     return true;
 }
 
-bool TextureAssetManager::Load(std::span<const std::string> paths, bool isExternal, asset_flags_type flags)
+auto TextureAssetManager::Load(std::span<const std::string> paths, std::span<const AssetSubtype> subtypes) -> bool
+{
+    NC_ASSERT(paths.size() == subtypes.size(), "Number of subtypes does not match number of asset paths");
```

Comment:
Since we're introducing a requirement for folder structure - we should add that to documentation

Reply:
blanking for some reason, what specific requirement are you referring to?